sql/src/main/java/io/crate/execution/engine/pipeline/Projectors.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.pipeline;
+
+import io.crate.breaker.RamAccountingContext;
+import io.crate.data.BatchIterator;
+import io.crate.data.Projector;
+import io.crate.data.Row;
+import io.crate.execution.dsl.projection.Projection;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.UUID;
+
+public final class Projectors {
+
+    private final ArrayList<Projector> projectors;
+    private final boolean independentScroll;
+
+    public Projectors(Collection<? extends Projection> projections,
+                      UUID jobId,
+                      RamAccountingContext ramAccountingContext,
+                      ProjectorFactory projectorFactory) {
+        boolean independentScroll = false;
+        this.projectors = new ArrayList<>(projections.size());
+        for (Projection projection : projections) {
+            if (projection.requiredGranularity().ordinal() > projectorFactory.supportedGranularity().ordinal()) {
+                continue;
```

Comment:
isn't this a programming error? should we at least log something/ add an assertion?

Reply:
Nope, see https://github.com/crate/crate/pull/7480 - I'll add some comments about it

Reply:
Pushed https://github.com/crate/crate/pull/7505/commits/8b68a8434dabd1aa970263828bb8a9e8eb0fb7d8